build.sh

```diff
@@ -221,10 +221,12 @@ BUILD_DIR="./${BUILD_TYPE}-build"
 
 rm -rf "${BUILD_DIR}"
 mkdir "${BUILD_DIR}"
+EXTRA_FLAGS=""
+[ "${BUILD_WITH_CLANG}" != "" ] && EXTRA_FLAGS="-DCMAKE_CXX_COMPILER=$LLVM_DIR/bin/clang++ -DLLVM_DIR=$LLVM_DIR"
```

Comment:
I would keep it as it is as some machines might not have clang installed.

Reply:
Added BUILD_WITH_CLANG

Reply:
Do we need to specify clang build? SVF is supposed to be built with g++ and others as long as it can compile

Reply:
I tried by using g++ which doesn't compile if I'm using android llvm-17, it only compiles clang, so I prefer to set BUILD_WITH_CLANG and use clang if perfered, it won't influence if using g++ by default.

Reply:
> supposed to be built with g++ an

There are the flags I used
```
export LLVM_DIR=/usr/src/clang-r487747
export LLVM_CONFIG=$LLVM_DIR/bin/llvm-config
export PATH=$LLVM_DIR/bin:$PATH
```
And build cmd
```
BUILD_WITH_CLANG=true source build.sh
```

Reply:
I see. Can this be configured in a nicer way and make sure you have the comments at the top of this file to explain the option too

Reply:
Why not call it compiler flag rather than extra flag and make it earlier so that people can see it together with other options for maintenance purposes